akka-actor-tests/src/test/scala/akka/actor/ReceiveTimeoutSpec.scala

```diff
@@ -290,5 +302,14 @@ class ReceiveTimeoutSpec extends AkkaSpec() {
       probe.expectMsg("restarting")
       probe.expectMsg(ReceiveTimeout)
     }
+
+    "Will cancel receive timeout task if stopped" in {
+      val probe = TestProbe()
+      val ref = system.actorOf(Props(new StoppingSelfActor(probe.ref)))
+      system.eventStream.subscribe(probe.ref, classOf[DeadLetter])
+      ref ! "Stop"
+      probe.expectMsg("stopping")
+      probe.expectNoMessage(3.seconds)
```

Comment:
Added a test

Reply:
Great with a test, but 3 seconds that this test case will always block is a looooong time. 

Can the test set a much shorter receive timeout (say 200ms) immediately before stopping on receiving the stop message, and then wait double that here instead?

Reply:
okay